pages-e2e/features/appMiddleware/middleware.test.ts

```diff
@@ -166,4 +166,19 @@ describe('App Middleware', () => {
 			text: 'json response from middleware',
 		});
 	});
+
+	test('middleware to be invoked once and only once', async ({ expect }) => {
```

Comment:
per request?

Reply:
yeah, I can see the ambiguity, sorry I'll add that 🙇

Reply:
text improved 🙂👍